docs/json-type-definition.md

```diff
@@ -313,6 +314,73 @@ Unlike JSON Schema, JTD does not allow to reference:
 
 Empty JTD schema defines the data instance that can be of any type, including JSON `null` (even if `nullable` member is not present). It cannot have any member other than `nullable` and `metadata`.
 
+## JTDSchemaType
+
+The type `JTDSchemaType` can be used to validate that the written schema matches the type you expect to validate. This type is strict such that if typescript compiles, you should require no further type guards. The downside of this is that the types that `JTDSchemaType` can verify are limited to the types that JTD can verify. If a type doesn't verify, `JTDSchemaType` should resolve to `never`, throwing an error when you try to assign to it. This means that types like `1 | 2 | 3`, or general untagged unions (outside of unions of string literals) cannot be used with `JTDSchemaType`.
+
+### Most Schemas
+
+Most straightforward types should work with `JTDSchemaType`, e.g.
+```typescript
+interface MyType {
+    num: number;
+    optionalStr?: string;
+    nullableEnum: "1" | "2";
```

Comment:
should it be `nullableEnum: "1" | "2" | null`
Or am I missing something?

Also, I'd use string strings here, for people accidentally not to think you can use numbers...